### PR TITLE
feat: Tests for Response service

### PR DESCRIPTION
**Pull request by rotimi-best**
## What does this PR do?

Implements jest test for response service.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://formbricks.com/clmyhzfrymr4ko00hycsg1tvx) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: packages/lib/response/tests/__mocks__/data.mock.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  include: typeof responseSelection;
+}>;
+type ResponseNoteMock = Prisma.ResponseNoteGetPayload<{
+  include: typeof responseNoteSelect;
+}>;
+type ResponsePersonMock = Prisma.PersonGetPayload<{
+  select: typeof responseSelection.person.select;
+}>;
+
+export const mockEnvironmentId = randomCuid2();
```
Comment (mattinannt) [inline packages/lib/response/tests/__mocks__/data.mock.ts:17]: @pandeymangg I think we shouldn't use random values in the tests since this would make the test non-deterministic:

---

Using random values in unit tests, like your `randBrowser()` and `randBoolean()` functions, can lead to non-deterministic tests, which might not be ideal in many testing scenarios. Here's a breakdown of the considerations:

### Non-Deterministic Tests: Drawbacks

1. **Reproducibility**: If a test fails due to a specific random value, it can be challenging to reproduce and debug the issue since the same value may not be generated again.
2. **Flakiness**: Tests might pass sometimes and fail at others, depending on the random values generated. This can lead to uncertainty about the stability and reliability of the code.
3. **Coverage Inconsistency**: Random values might not consistently cover all the edge cases or paths in the code, leading to inconsistent test coverage.

### Deterministic Tests: Advantages

1. **Reproducibility**: Tests produce the same results every time, making it easier to identify and fix issues.
2. **Reliability**: Consistent outcomes increase confidence in the test results and the code's stability.
3. **Complete Coverage**: You can design tests to cover all relevant cases and edge cases, ensuring comprehensive testing.

### When to Use Randomness

Using randomness in tests can be beneficial in certain scenarios:

1. **Stress Testing**: To test how the system behaves under varied and unpredictable inputs.
2. **Fuzz Testing**: To identify potential crashes or unexpected behavior in the system.
3. **Large Data Sets**: When dealing with huge datasets, random sampling might be necessary for practicality.

### Best Practices

- **Controlled Randomness**: If you use random values, consider using a seeded random number generator. This way, the "random" values are the same across different test runs, maintaining some level of determinism.
- **Specific Test Cases for Edge Conditions**: Write separate deterministic tests for edge cases and specific conditions you want to ensure are always tested.
- **Monitoring Flaky Tests**: If you use randomness, monitor for flakiness in your test suite, and investigate tests that show inconsistent results.

### Conclusion

In your context, if the random values are not creating flakiness or missing coverage, they might be acceptable. However, generally, it's better to have deterministic tests, especially for unit testing, where you're testing specific functions or components under known conditions. For broader system testing or in specific scenarios like stress or fuzz testing, randomness can be more appropriate.

**Review by mattinannt — COMMENTED** [review]
@pandeymangg Looks pretty good :-)
How do we best run this in Github Action CI? We already have the action that runs the formbricks-js tests on `pnpm test`. It would be great if we can run all unit tests (including formbricks-js tests which are currently disabled / commented out) by running `pnpm test` in the project root.

File: packages/lib/response/tests/__mocks__/data.mock.ts (hunk 1 above)
Comment (pandeymangg) [reply packages/lib/response/tests/__mocks__/data.mock.ts:17]: okay, should we then hard code these values?

File: packages/lib/response/tests/constants.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+export function randUuid(): string {
```
Comment (mattinannt) [inline packages/lib/response/tests/constants.ts:1]: the function name doesn't really make sense now since we don't have random values. can we maybe also change this to constants?

**Review by mattinannt — COMMENTED** [review]
@pandeymangg Thanks for the changes 💪; unfortunately there are a few things still confusing about the function names. Please update the function names (or use constants if that makes sense), then we can merge 😊

**Review by mattinannt — APPROVED** [review]
@rotimi-best @pandeymangg Thanks a lot for the unit testing infrastructure for services! Will merge it into main now 😊💪🚀
